### PR TITLE
GPII-3197: Adding allow_upgrade chart config flag

### PR DESCRIPTION
**Pull request by natarajaya**
This should prevent annoying 60 seconds of retries during `gpii-dataloader` chart upgrade.

### CODE REVIEW

File: aws/modules/deploy/charts/config.yml [hunk 1]
```diff
@@ -7,11 +7,12 @@
...
 #
-# Every chart should have the following attributes:
+# Every chart may have the following attributes:
 #
 # beautiful-chart:
-#  release_name: release-name-to-pass-to-helm
-#  release_namespace: release-namespace-to-pass-to-helm
+#  release_name: release-name-to-pass-to-helm              – optional, defaults to chart name
+#  release_namespace: release-namespace-to-pass-to-helm    - optional, defaults to "default" namespace
+#  allow_upgrade: false                                    - optional, default is false
```
Comment (amatas) [inline aws/modules/deploy/charts/config.yml]: Sorry, @natarajaya but the logic confuses me a bit:

if default is false, it means that by default all the charts are not allowed to be upgraded, unless you set it to true but then in deploy.rake, if the value is true is when the helm is updated. Wouldn't it be clearer if the default value is `true`?

File: aws/rakefiles/deploy.rake [hunk 2]
```diff
@@ -165,16 +165,25 @@ task :install_charts => [:configure_kubectl, :generate_modules, :setup_system_co
     else
       chart_namespace = 'default'
     end
+    if attrs && !attrs['allow_upgrade'].nil? && attrs['allow_upgrade'] == false
+      allow_upgrade = false
+    else
+      allow_upgrade = true
+    end
```
Comment (amatas) [inline aws/rakefiles/deploy.rake]: Perhaps this if block can be replaced by:
```
allow_upgrade = attrs.fetch('allow_upgrade', true)
```
or something similar in order to see clearly that by default the value is **true**.

File: aws/modules/deploy/charts/config.yml (hunk 1 above)
Comment (natarajaya) [reply aws/modules/deploy/charts/config.yml]: Yep, mistake in the comment – default is `true`.

File: aws/rakefiles/deploy.rake (hunk 2 above)
Comment (natarajaya) [reply aws/rakefiles/deploy.rake]: Nice suggestion!

File: aws/rakefiles/deploy.rake [hunk 3]
```diff
@@ -154,27 +154,24 @@ task :install_charts => [:configure_kubectl, :generate_modules, :setup_system_co
...
-    end
-    if attrs && attrs['release_namespace']
-      chart_namespace = attrs['release_namespace']
-    else
-      chart_namespace = 'default'
-    end
-
+    # Extracting release attributes
+    attrs ||= {}
+    chart_name = attrs.fetch('release_name', chart)
```
Comment (mrtyler) [inline aws/rakefiles/deploy.rake]: The regex removing the leading digits is no longer needed?

Comment (natarajaya) [reply aws/rakefiles/deploy.rake]: Yes, we no longer naming charts with numerical prefixes, since we introduced yaml charts config.

**Comment by mrtyler** [comment]
LGTM

**Pull request merged**